sdk/go/pulumi/context.go

```diff
@@ -277,6 +277,36 @@ func (ctx *Context) ReadResource(
 		options.Parent = ctx.stack
 	}
 
+	// Before anything else, if there are transformations registered, give them a chance to run to modify the
+	// user-provided properties and options assigned to this resource.
+	transformations := options.Transformations
+	if options.Parent != nil {
```

Comment:
How are stack transformations handled? Does the engine automatically apply them to all resources, or do we need to copy them to children resources as well?

Reply:
The stack is the parent of all un-parented resources. See https://www.pulumi.com/docs/intro/concepts/programming-model/#parent.